app/src/main/java/com/example/inharmony/ChatAdapter.java

```diff
@@ -0,0 +1,204 @@
+        ImageView imageOther;
+        TextView body;
+        TextView name;
+        TextView tvIncomingTrack;
+        ImageView ivIncomingTrack;
+        ImageView ivPlayIncoming;
+
+          public IncomingMessageViewHolder(View itemView) {
+            super(itemView);
+            imageOther = itemView.findViewById(R.id.ivProfileOther);
+            body = itemView.findViewById(R.id.tvOtherBody);
+            name = itemView.findViewById(R.id.tvSender);
+            tvIncomingTrack = itemView.findViewById(R.id.tvIncomingTrack);
+            ivIncomingTrack = itemView.findViewById(R.id.ivIncomingTrack);
+            ivPlayIncoming = itemView.findViewById(R.id.ivPlayIncoming);
+        }
+
+        @Override
+        void bindMessage(Message message) {
+            ParseUser user = message.getSender();
+            try {
+                name.setText(user.fetch().getUsername());
+                if (user.get("profilePic") != null) {
+                    ParseFile image = (ParseFile) user.get("profilePic");
+                    Glide.with(mContext).load(image.getUrl()).into(imageOther);
+                }
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+
+            if (message.getBody() == null) {
+                body.setVisibility(View.GONE);
+            } else {
+                body.setVisibility(View.VISIBLE);
+                body.setText(message.getBody());
+            }
+
+            if (message.getTrackId() == null) {
+                tvIncomingTrack.setVisibility(View.GONE);
+                ivPlayIncoming.setVisibility(View.GONE);
+                ivIncomingTrack.setVisibility(View.GONE);
+            } else {
+                tvIncomingTrack.setVisibility(View.VISIBLE);
+                ivPlayIncoming.setVisibility(View.VISIBLE);
+                ivIncomingTrack.setVisibility(View.VISIBLE);
+                SpotifyApi spotifyApi = new SpotifyApi();
+                spotifyApi.setAccessToken(token);
+                SpotifyService service = spotifyApi.getService();
+                LoadChatTrackTask loadChatTrackTask = new LoadChatTrackTask(itemView, token, service, false);
+                loadChatTrackTask.execute(message);
+
+
+            }
+        }
+
+    }
+
+    public class OutgoingMessageViewHolder extends MessageViewHolder {
+        ImageView imageMe;
+        TextView body;
+        TextView tvOutgoingTrack;
+        ImageView ivOutgoingTrack;
+        ImageView ivPlayOutgoing;
+
+        public OutgoingMessageViewHolder(View itemView) {
+            super(itemView);
+            imageMe = itemView.findViewById(R.id.ivProfileMe);
+            body = itemView.findViewById(R.id.tvBody);
+            tvOutgoingTrack = itemView.findViewById(R.id.tvOutgoingTrack);
+            ivOutgoingTrack = itemView.findViewById(R.id.ivOutgoingTrack);
+            ivPlayOutgoing = itemView.findViewById(R.id.ivPlayOutgoing);
```

Comment:
nit: we could assign a new variable `id` to `R.id` and reference that in the `findViewById()`, etc. args, rather than referencing R.id each time (that is technically a lookup we can remove).